diff --git a/src/functions/Helpers.ts b/src/functions/Helpers.ts
index e04486540494891ab07ec130b686dc4acddf2d0c..265e6ac11439276a1c52c222dfc4c50daf1689ae 100644
--- a/src/functions/Helpers.ts
+++ b/src/functions/Helpers.ts
@@ -77,7 +77,8 @@ export function getNativeNodeHandle(nativeRef: React.Component){
   const nodeHandle = findNodeHandle(nativeRef);
 
   if(nodeHandle == null){
-    throw new Error('Unable to get the node handle for the native ref.');
+    return 0
+    // throw new Error('Unable to get the node handle for the native ref.');
   };
 
   return nodeHandle;
