diff --git a/node_modules/@sentry/react-native/ios/RNSentry.h b/node_modules/@sentry/react-native/ios/RNSentry.h
index 6f7df58..84b50c9 100644
--- a/node_modules/@sentry/react-native/ios/RNSentry.h
+++ b/node_modules/@sentry/react-native/ios/RNSentry.h
@@ -1,7 +1,7 @@
 #if __has_include(<React/RCTBridge.h>)
 #import <React/RCTBridge.h>
 #else
-#import "RCTBridge.h"
+#import <React/RCTBridge.h>
 #endif
 
 @interface RNSentry : NSObject <RCTBridgeModule>
diff --git a/node_modules/@sentry/react-native/ios/RNSentry.m b/node_modules/@sentry/react-native/ios/RNSentry.m
index 32a6087..003a001 100644
--- a/node_modules/@sentry/react-native/ios/RNSentry.m
+++ b/node_modules/@sentry/react-native/ios/RNSentry.m
@@ -3,7 +3,7 @@
 #if __has_include(<React/RCTConvert.h>)
 #import <React/RCTConvert.h>
 #else
-#import "RCTConvert.h"
+#import <React/RCTConvert.h>
 #endif
 
 #import <Sentry/Sentry.h>
diff --git a/node_modules/@sentry/react-native/react-native.config.js b/node_modules/@sentry/react-native/react-native.config.js
index f85fa01..053999b 100644
--- a/node_modules/@sentry/react-native/react-native.config.js
+++ b/node_modules/@sentry/react-native/react-native.config.js
@@ -1,18 +1,10 @@
 module.exports = {
   dependency: {
     platforms: {
-      ios: {
-        sharedLibraries: ['libz']
-      },
+      ios: {},
       android: {
         packageInstance: 'new RNSentryPackage()'
       }
-    },
-    hooks: {
-      postlink:
-        'node node_modules/@sentry/wizard/dist/bin.js -i reactNative -p ios android',
-      postunlink:
-        'node node_modules/@sentry/wizard/dist/bin.js -i reactNative -p ios android --uninstall'
     }
   }
 };
\ No newline at end of file
