diff --git a/node_modules/@types/react-native-share-menu/index.d.ts b/node_modules/@types/react-native-share-menu/index.d.ts
index f52822c..ee98565 100755
--- a/node_modules/@types/react-native-share-menu/index.d.ts
+++ b/node_modules/@types/react-native-share-menu/index.d.ts
@@ -5,11 +5,9 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // Minimum TypeScript Version: 3.7
 
-export interface ShareData {
-    mimeType: string;
-    data: string | string[];
-    extraData?: object | undefined;
-}
+export type ShareData = {
+    data: {mimeType: string; data: string}[];
+} | {mimeType: string; data: string | string[]}
 
 export type ShareCallback = (share?: ShareData) => void;
 
@@ -28,7 +26,7 @@ interface ShareMenuReactView {
     dismissExtension(error?: string): void;
     openApp(): void;
     continueInApp(extraData?: object): void;
-    data(): Promise<{mimeType: string, data: string}>;
+    data(): Promise<{data: {mimeType: string; data: string}[]}>;
 }
 
 export const ShareMenuReactView: ShareMenuReactView;
