diff --git a/node_modules/expo-av/ios/EXAV/EXAudioSessionManager.m b/node_modules/expo-av/ios/EXAV/EXAudioSessionManager.m
index 81dce13..8664b90 100644
--- a/node_modules/expo-av/ios/EXAV/EXAudioSessionManager.m
+++ b/node_modules/expo-av/ios/EXAV/EXAudioSessionManager.m
@@ -168,9 +168,6 @@ - (void)moduleDidBackground:(id)backgroundingModule
   // compact doesn't work, that's why we need the `|| !pointer` above
   // http://www.openradar.me/15396578
   [_foregroundedModules compact];
-
-  // Any possible failures are silent
-  [self _updateSessionConfiguration];
 }
 
 - (void)moduleDidForeground:(id)module
