diff --git a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt
index 55e08c2..dee876b 100644
--- a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt
+++ b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt
@@ -165,7 +165,7 @@ open class ScreenContainer<T : ScreenFragment>(context: Context?) : ViewGroup(co
             try {
                 FragmentManager.findFragment<Fragment>(rootView).childFragmentManager
             } catch (ex: IllegalStateException) {
-                throw IllegalStateException("Failed to find fragment for React Root View")
+                context.supportFragmentManager
             }
         }
     }
